File: 96. Unique Binary Search Trees.java
class Solution {
    public int numTrees(int n) {
        int arr[] = new int[n+1];
        arr[0] = 1;
        arr[1] = 1;
        for(int i = 2; i < n+1; i++){
            for(int j =1; j < i+1;j++){
                arr[i]+=arr[j-1]*arr[i-j];
            }
        }
        return arr[n];
    }
}

File: Dynamic Programming/96. Unique Binary Search Trees.py
class Solution(object):
    def numTrees(self, n):
        """
        :type n: int
        :rtype: int
        """
        arr = [0] * (n+1)
        arr[0] = 1
        arr[1] = 1
        for i in range(2,n+1):
            for j in range(1,i+1):
                arr[i]+=arr[j-1]*arr[i-j]
        return arr[n]  